modules/redpanda/redpanda_test.go

```diff
@@ -573,3 +574,58 @@ func TestRedpandaListener_NoNetwork(t *testing.T) {
 
 	require.Contains(t, err.Error(), "container must be attached to at least one network")
 }
+
+func TestRedpandaBootstrapConfig(t *testing.T) {
+	ctx := context.Background()
+
+	container, err := redpanda.RunContainer(ctx,
+		redpanda.WithEnableWasmTransform(),
+		// These configs would require a restart if applied to a live Redpanda instance
+		redpanda.WithBootstrapConfig("data_transforms_per_core_memory_reservation", 33554432),
+		redpanda.WithBootstrapConfig("data_transforms_per_function_memory_limit", 16777216),
+	)
+	require.NoError(t, err)
+
+	httpCl := &http.Client{
+		Timeout: 5 * time.Second,
+		Transport: &http.Transport{
+			ForceAttemptHTTP2:   true,
+			TLSHandshakeTimeout: 10 * time.Second,
+		},
+	}
+	adminAPIUrl, err := container.AdminAPIAddress(ctx)
+	require.NoError(t, err)
+
+	{
+		// Check that the configs reflect specified values
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("%s/v1/cluster_config", adminAPIUrl), nil)
+		require.NoError(t, err)
+		resp, err := httpCl.Do(req)
+		require.NoError(t, err)
+		defer resp.Body.Close()
+		require.Equal(t, http.StatusOK, resp.StatusCode)
+		var data map[string]any
+		err = json.NewDecoder(resp.Body).Decode(&data)
+		require.NoError(t, err)
+		reservation := int(data["data_transforms_per_core_memory_reservation"].(float64))
+		assert.Equal(t, 33554432, reservation)
+		pf_limit := int(data["data_transforms_per_function_memory_limit"].(float64))
+		assert.Equal(t, 16777216, pf_limit)
+	}
+
+	{
+		// Check that no restart is required. i.e. that the configs were applied via bootstrap config
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("%s/v1/cluster_config/status", adminAPIUrl), nil)
+		require.NoError(t, err)
+		resp, err := httpCl.Do(req)
+		require.NoError(t, err)
+		defer resp.Body.Close()
+		require.Equal(t, http.StatusOK, resp.StatusCode)
+		var data []map[string]any
+		err = json.NewDecoder(resp.Body).Decode(&data)
+		require.NoError(t, err)
+		assert.Len(t, data, 1)
```

Comment:
suggestion: This is actually a good example of why require is preferable to assert.

If the length here was zero the assert would mark the test as failed, but would allow it continue, the following line would then panic as it indexed into an empty slice.

Reply:
good catch, yeah. need to have it beaten into my brain that `assert` is the non-halting version...